File: cinder/Entrainment/src/ViewNormal.cpp
//
//  ViewNormal.cpp
//  Entrainment03
//
//  Created by Yi-Wen Lin on 2015/7/24.
//
//

#include "ViewNormal.h"
#include "GlobalSettings.h"
#include "MeshUtils.h"

using namespace ci;
using namespace ci::app;
using namespace std;

ViewNormal::ViewNormal() : View("shaders/copy.vert", "shaders/normal.frag") {
    _init();
}

ViewNormal::ViewNormal(string vsPath, string fsPath) : View(vsPath, fsPath) {
    _init();
}


void ViewNormal::_init() {
    mesh = bongiovi::MeshUtils::createPlane(2, 1);
}

void ViewNormal::render(gl::TextureRef texture) {
    render(*texture);
}

void ViewNormal::render(gl::Texture texture) {
    shader->bind();
    shader->uniform("texture", 0);
    shader->uniform("scale", _scale);
    texture.bind();
    gl::draw(mesh);
    texture.unbind();
    shader->unbind();
}

File: webgl/src/js/terrain/ViewNormal.js
// ViewNormal.js

var GL = bongiovi.GL;
var gl;
var glslify = require("glslify");

function ViewNormal(mScale) {
	this._scale = mScale === undefined ? 1.0 : mScale;
	console.log('Scale ', this._scale);
	bongiovi.View.call(this, null, glslify("../../shaders/normal.frag"));
}

var p = ViewNormal.prototype = new bongiovi.View();
p.constructor = ViewNormal;


p._init = function() {
	gl = GL.gl;
	this.mesh = bongiovi.MeshUtils.createPlane(2, 2, 1);
};

p.render = function(texture) {
	this.shader.bind();
	this.shader.uniform("texture", "uniform1i", 0);
	this.shader.uniform("scale", "uniform1f", this._scale);
	texture.bind(0);
	GL.draw(this.mesh);
};

module.exports = ViewNormal;